sdk/cargo-build-sbf/src/main.rs

```diff
@@ -933,6 +1016,13 @@ fn main() {
                 .takes_value(false)
                 .help("Run without accessing the network"),
         )
+        .arg(
+            Arg::new("tools_version")
+                .long("tools-version")
+                .value_name("STRING")
```

Comment:
Let's add a clap validator to this as well, like
https://github.com/solana-labs/solana/blob/3aa0a005f942406968a29c81822564ec994959c9/install/src/lib.rs#L151

for example it looks like if I passed in `--tools-version X`, we'd panic on:
```rust
let requested_semver = semver::Version::parse(&normalized_requested[1..]).unwrap();
```